### PR TITLE
Add package build and publish to CI

### PR DESCRIPTION
**Pull request by abigailalexander**
I updated package.json to bundle the built `/dist` directory in the npm published package, instead of the unbuilt `/src` directory. Added CI to automate the publishing of the package to the registry.

The latest version of plugin-scaffolder-backend added some properties to ActionContext causing `yarn tsc` to fail because of `copier.test.ts`. I have included createMockActionContext in `copier.test.ts` to help restore and maintain compatibility in the long run with structural changes to action context (mentioned in [changelog](https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/CHANGELOG.md) for plugin-scaffolder-backend).

### CODE REVIEW

**Comment by abigailalexander** [comment]
For the npm publishing CI to work, a token needs to be added to this repo (I don't have permission to do so).

File: .github/workflows/npm-release.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+on:
+  push:
+    tags:
+    - '*'
+
+jobs:
+  build: 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (garryod) [inline .github/workflows/npm-release.yml]: Usually worth using fully qualified versions for these

**Review by garryod — APPROVED** [review]
Lgtm (once token is added obviously)

File: .github/workflows/npm-release.yml (hunk 1 above)
Comment (coretl) [reply .github/workflows/npm-release.yml]: I'm going to disagree with this one, what I normally do:
- Anything in the `actions` or another reputable org like `pypa` references the major version (e.g. v2)
- Anything else pins to the SHA of the tag with a comment saying which version it is

The reasoning is that we don't want the noise of constantly updating actions versions (even if it's autocommitted), and we trust GitHub and other large orgs not to break us, but third party actions from a single developer we do not trust so pin to a single version.

What's your thinking on this @garryod ?

**Comment by coretl** [comment]
> For the npm publishing CI to work, a token needs to be added to this repo (I don't have permission to do so).

You are now an admin on this repo so you can do that

**Comment by abigailalexander** [comment]
Thanks for adding me as an admin @coretl. I'll leave the github actions as they are for now and merge the PR, so I can make a release and update dev-portal. If Garry thinks specifying only major version is better, I'll open a new PR later to change the action versions.

**Pull request merged**